src/dmd/cli.d

```diff
@@ -169,14 +169,30 @@ struct Usage
         Option("c",
             "compile only, do not link"
         ),
+        Option("check=[assert|bounds|in|invariant|out|switch][=[on|off]]",
+            "Enable or disable specific checks",
+            `Overrides default, -boundscheck, and -release options to enable or disable specific checks.
```

Comment:
It currently overrides -boundscheck only if it is specified after it on the command line. I'd rather leave `bounds` out here, it already has two options and another setting `safeonly`.

Reply:
We actually have `boundscheck` and `noboundscheck`. I'd like to deprecate them and just go with the more general `check`. I also want to add `safe` as a third option to `check`, so that all the checks are handled the same way with one switch.

Reply:
> We actually have `boundscheck` and `noboundscheck`. I'd like to deprecate them and just go with the more general `check`. I also want to add `safe` as a third option to `check`, so that all the checks are handled the same way with one switch.

If you want to go this route, I might suggest considering adding back the `all` and `none` values present in the DIP.

Reply:
On the other hand, what you said afterwards contradicts this idea:

> The idea is that changing one setting with -check has no effect on any other settings. It's designed so that each setting can be controlled independently of the others. Therefore, if you set check=in, then the in contracts are run, but if assert is turned off, then they'll be vacuous.

Reply:
> It currently overrides -boundscheck only if it is specified after it on the command line.

Fixed.